src/unstract/adapters/vectordb/milvus/src/milvus.py

```diff
@@ -48,36 +49,43 @@ def get_json_schema() -> str:
         return schema
 
     def get_vector_db_instance(self) -> VectorStore:
+        return self._vector_db_instance
+
+    def _get_vector_db_instance(self) -> VectorStore:
         try:
-            self.collection_name = VectorDBHelper.get_collection_name(
-                self.config.get(VectorDbConstants.VECTOR_DB_NAME),
-                self.config.get(VectorDbConstants.EMBEDDING_DIMENSION),
+            self._collection_name = VectorDBHelper.get_collection_name(
+                self._config.get(VectorDbConstants.VECTOR_DB_NAME),
+                self._config.get(VectorDbConstants.EMBEDDING_DIMENSION),
             )
-            dimension = self.config.get(
+            dimension = self._config.get(
                 VectorDbConstants.EMBEDDING_DIMENSION,
                 VectorDbConstants.DEFAULT_EMBEDDING_SIZE,
             )
             vector_db: VectorStore = MilvusVectorStore(
-                uri=self.config.get(Constants.URI, ""),
-                collection_name=self.collection_name,
-                token=self.config.get(Constants.TOKEN, ""),
+                uri=self._config.get(Constants.URI, ""),
+                collection_name=self._collection_name,
+                token=self._config.get(Constants.TOKEN, ""),
                 dim=dimension,
             )
             if vector_db is not None:
-                self.client = vector_db.client
+                self._client = vector_db.client
             return vector_db
         except Exception as e:
             raise AdapterError(str(e))
 
     def test_connection(self) -> bool:
-        self.config[VectorDbConstants.EMBEDDING_DIMENSION] = (
+        self._config[VectorDbConstants.EMBEDDING_DIMENSION] = (
             VectorDbConstants.TEST_CONNECTION_EMBEDDING_SIZE
         )
         vector_db = self.get_vector_db_instance()
         test_result: bool = VectorDBHelper.test_vector_db_instance(
             vector_store=vector_db
         )
         # Delete the collection that was created for testing
-        if self.client is not None:
-            self.client.drop_collection(self.collection_name)
+        if self._client is not None:
+            self._client.drop_collection(self._collection_name)
         return test_result
+
+    def close(self, **kwargs: Any) -> None:
```

Comment:
@gaya3-zipstack  from where all are we calling this close function?

Reply:
I need to integrate close() with vector db usages in client across. Yet to take this up! Since we see some timeout issues, thought of adding this hook for close. But not integrated yet